src/index.js

```diff
@@ -55,59 +55,62 @@ const createRunner = ({ runtime = "worker_threads" } = {}) =>
      * @param {*} onResult
      * @param {*} onFailure
      */
-    runTests(tests, watcher, onStart, onResult, onFailure) {
+    async runTests(tests, watcher, onStart, onResult, onFailure) {
       const pool = this._pool;
       const { updateSnapshot, testNamePattern, maxWorkers } = this._config;
-
-      if (!this._runInBand && this._isProcessRunner) {
-        return pMap(
-          tests,
-          test => {
-            onStart(test);
-
-            return pool.run({ test, updateSnapshot, testNamePattern }).then(
-              result => void onResult(test, result),
-              error => void onFailure(test, error),
-            );
-          },
-          { concurrency: maxWorkers },
-        ).finally(async () => {
-          for (const { process } of pool.threads) {
-            // Use `process.disconnect()` instead of `process.kill()`, so we can collect coverage
-            // See https://github.com/nicolo-ribaudo/jest-light-runner/issues/90#issuecomment-2812473389
-            // Only override the first call https://github.com/tinylibs/tinypool/blob/dbf6d74282dd6031df8fc5c7706caef66b54070b/src/runtime/process-worker.ts#L61
-            const originalKill = process.kill;
-            process.kill = signal => {
-              if (!signal) {
-                process.disconnect();
-                process.kill = originalKill;
-                return;
-              }
-              return originalKill.call(process, signal);
-            };
-          }
-
-          await pool.destroy();
-        });
-      }
-
-      return Promise.all(
-        tests.map(test => {
+      const isProcessRunner = !this._runInBand && this._isProcessRunner;
+
+      let run;
+      if (isProcessRunner) {
+        run = async test => {
+          await onStart(test);
+          return pool.run({ test, updateSnapshot, testNamePattern });
+        };
+      } else {
+        run = test => {
           const mc = new MessageChannel();
           mc.port2.onmessage = () => onStart(test);
           mc.port2.unref();
```

Comment:
Since we used `p-limit`, we should be able to remove them now.

Reply:
I wasn't sure about this, is it added to get precise timing?

@nicolo-ribaudo

Reply:
Anyway, it should be improved, I often got screen like this...

![image](https://github.com/user-attachments/assets/56d70b0f-8b08-45f5-bc47-e090d2cb6d5a)

Reply:
https://github.com/nicolo-ribaudo/jest-light-runner/blob/42c110f6b5dff98a87ddf89f82ff2015167bc292/src/worker-runner.js#L59
I guess this is to show `running` correctly.
Yeah, it would be great if this could be improved.

Reply:
Yeah this was to know when the test actually starts, rather than when we just enqueue it in the worker/process.

Reply:
So, should we keep it?

Reply:
I think with `p-limit` we don't need it, because we know when the work actually starts, right? Because `p-limit` runs the callback to actually start the work, unlike the old pool-based implementation that doesn't tell us when the work gets processed.

Reply:
Released https://github.com/nicolo-ribaudo/jest-light-runner/releases/tag/v0.7.5 if you want upgrade.